dora/core/server/worker/src/main/java/alluxio/worker/dora/CachingDoraMetaStore.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.dora;
+
+import alluxio.collections.LockPool;
+import alluxio.concurrent.LockMode;
+import alluxio.conf.Configuration;
+import alluxio.conf.PropertyKey;
+import alluxio.metadata.caching.Cache;
+import alluxio.metadata.caching.CacheConfiguration;
+import alluxio.metrics.MetricKey;
+import alluxio.proto.meta.DoraMeta.FileStatus;
+import alluxio.resource.LockResource;
+import alluxio.resource.RWLockResource;
+import alluxio.util.ConfigurationUtils;
+
+import com.google.common.base.Preconditions;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * The cached dora metadata Store.
+ */
+public class CachingDoraMetaStore implements DoraMetaStore {
```

Comment:
So here we created a caching meta store object. I discussed with @huanghua78 today and we realized that we probably might not have to have this 2 layer system. 

The historical background why we introduced a caching inode store in alluxio 2.x was because the performance of rocksdb only inode store wasn't good enough. But when it comes to the metadata store on dora workers, a couple of conditions have been changed since then:
1. more users use nvme as storage instead of HDD
2. each dora worker only persists a small portion of metadata, unlike master, which persists all metadata. The QPS on each worker will be much smaller than the previous master QPS. 

so using a rocksdb only meta store might simplify the implementation and eliminates more performance issues. But I'm not very sure on that and would like to hear from @beinan

Reply:
Based on the stress test from @LuQQiu , using a large in-heap cache might cause GC issues,  so I lean towards to using rockdb only solution if there is no significant performance drop.

Reply:
yup our profiling result demonstrated the same so we removed this